azure-configurationservice/azure/configuration/azure_configuration_client_abstract.py

```diff
@@ -0,0 +1,301 @@
+        # type: (str, str, str, dict) -> ConfigurationSetting
+
+        """Delete a ConfigurationSetting if it exists
+
+        :param key: key used to identify the ConfigurationSetting
+        :type key: str
+        :param label: label used to identify the ConfigurationSetting
+        :type label: str
+        :param etag: check if the ConfigurationSetting is changed. Set None to skip checking etag
+        :type etag: str
+        :param kwargs: if "headers" exists, its value (a dict) will be added to the http request
+        :type kwargs: dict
+        :return: The deleted ConfigurationSetting returned from the service, or None if it doesn't exist.
+        :rtype: :class:`ConfigurationSetting`
+        :raises: :class:`ResourceNotFoundError`, :class:`ResourceModifiedError`, :class:`HttpRequestError`
+
+        """
+        custom_headers = CaseInsensitiveDict(kwargs.get("headers"))
+        if etag:
+            custom_headers["if-match"] = AzureConfigurationClientAbstract.quote_etag(
+                etag
+            )
+        return self._impl.delete_configuration_setting(
+            key=key,
+            label=label,
+            headers=custom_headers,
+            error_map={
+                404: ResourceNotFoundError,  # 404 doesn't happen actually. return None if no match
+                412: ResourceModifiedError,
+            },
+        )
+
+    @abstractmethod
+    def list_revisions(
+        self, labels=None, keys=None, accept_date_time=None, fields=None, **kwargs
+    ):
+        # type: (list, list, datetime, list, dict) -> Paged
+
+        """
+        Find the ConfigurationSetting revision history.
+
+        :param labels: filter results based on their label. '*' can be
+         used as wildcard in the beginning or end of the filter
+        :type labels: list[str]
+        :param keys: filter results based on their keys. '*' can be
+         used as wildcard in the beginning or end of the filter
+        :type keys: list[str]
+        :param accept_date_time: filter out ConfigurationSetting created after this datetime
+        :type accept_date_time: datetime
+        :param fields: specify which fields to include in the results. Leave None to include all fields
+        :type fields: list[str]
+        :param dict kwargs: if "headers" exists, its value (a dict) will be added to the http request header
+        :return: An iterator of :class:`ConfigurationSetting`
+        :rtype: :class:`Paged`
+        :raises: :class:`HttpRequestError`
+
+        """
+
+        labels = AzureConfigurationClientAbstract.escape_and_tolist(labels)
+        keys = AzureConfigurationClientAbstract.escape_and_tolist(keys)
+        return self._impl.list_revisions(
+            label=labels,
+            key=keys,
+            fields=fields,
+            accept_date_time=accept_date_time,
+            headers=kwargs.get("headers"),
+        )
+
+    @staticmethod
```

Comment:
Why is this a static method?

Reply:
The method has nothing to do with the class members (variables nor metnhods), so it should be either static or @classmethod. And it doesn't use cls, so it's static

Reply:
It uses quote_etag. It just hard-codes the class name. The way the code is structured, it becomes difficult to mock and test.

Reply:
quote_etag is also static. They both are static.

Reply:
When I say "members", I mean instance variables/methods, not the class level or static ones.

Reply:
I rarely use static functions in Python. Since you have module level functions, there is rarely a need unless you need additional grouping.